### ISSUE TITLE
Use transitions.extensions.GraphMachine to allow making state transition diagrams

### ISSUE THREAD

**Issue by kimberscott** [lookit/lookit-api#494]
**TL;DR**: Go back to using the GraphMachine extension for transitions so we can make transition diagrams for studies.

**Narrative**
"As a Developer, it would be helpful to be able to automatically generate state transition diagrams when considering any changes to the Study workflow. "

**Implementation Notes**

Kim switched from using `transitions.extensions.GraphMachine` back to `transitions.Machine` for the Study model state machine as part of upgrading to Django 3.0, because the former caused problems of the following form (same error on all tests):

```
ERROR: testDeletePreviewData (studies.tests.ResponseViewsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/kms/lookit-api/studies/tests.py", line 48, in setUp
    organization=self.org,
  File "<string>", line 6, in get
  File "/Users/kms/.local/share/virtualenvs/lookit-api-TotXiiub/lib/python3.6/site-packages/django_dynamic_fixture/__init__.py", line 148, in _get
    return d.get(model, ddf_lesson=ddf_lesson, **kwargs)
  File "/Users/kms/.local/share/virtualenvs/lookit-api-TotXiiub/lib/python3.6/site-packages/django_dynamic_fixture/ddf.py", line 644, in get
    instance = self.new(model_class, ddf_lesson=ddf_lesson, **kwargs)
  File "/Users/kms/.local/share/virtualenvs/lookit-api-TotXiiub/lib/python3.6/site-packages/django_dynamic_fixture/ddf.py", line 536, in new
    instance = model_class()
  File "/Users/kms/lookit-api/studies/models.py", line 155, in __init__
    after_state_change="_finalize_state_change",
  File "/Users/kms/.local/share/virtualenvs/lookit-api-TotXiiub/lib/python3.6/site-packages/transitions/extensions/diagrams.py", line 150, in __init__
    _super(GraphMachine, self).__init__(*args, **kwargs)
  File "/Users/kms/.local/share/virtualenvs/lookit-api-TotXiiub/lib/python3.6/site-packages/transitions/extensions/markup.py", line 28, in __init__
    super(MarkupMachine, self).__init__(*args, **kwargs)
  File "/Users/kms/.local/share/virtualenvs/lookit-api-TotXiiub/lib/python3.6/site-packages/transitions/core.py", line 574, in __init__
    self.add_model(model)
  File "/Users/kms/.local/share/virtualenvs/lookit-api-TotXiiub/lib/python3.6/site-packages/transitions/extensions/diagrams.py", line 216, in add_model
    _ = mod.get_graph(title=self.title, force_new=True)  # initialises graph
  File "/Users/kms/.local/share/virtualenvs/lookit-api-TotXiiub/lib/python3.6/site-packages/transitions/extensions/diagrams.py", line 179, in _get_graph
    self.model_graphs[model] = grph
  File "/Users/kms/.local/share/virtualenvs/lookit-api-TotXiiub/lib/python3.6/site-packages/django/db/models/base.py", line 535, in __hash__
    raise TypeError("Model instances without primary key value are unhashable")
TypeError: Model instances without primary key value are unhashable
```

The path to fixing this is reasonably straightforward - just deferring to focus on higher-priority stuff than being able to make state diagrams. Rico's advice:

> so the problem is that the __init__ override we have on the Study model adds this StateMachine object after initializing the object. The problem is that Django does not add a PK to a model instance until it gets it from the DB, which happens during save()
> 
> and the graphing mechanism that gets invoked somewhere in that Machine constructor needs a pk
> 
> I would avoid calling save() in the constructor - at least from what I remember that can cause problems. At the very least I would poke around the interwebs and make sure there’s no immediately obvious issues
> 
> It appears that Dawn followed the procedure here https://github.com/pytransitions/transitions/issues/111#issuecomment-227762550
> 
> I would recommend seeing what they do here: https://github.com/PrimarySite/django-transitions - small codebase, and it looks like they use a pretty slick mixin pattern that registers a state machine at the class level https://django-transitions.readthedocs.io/en/latest/quickstart.html#statemachine-mixin. That would also save memory since you aren’t creating a state machine for each model instance

Indeed, in `studies/models.py`, the dumb approach of adding `self.save()` goes like this:

```
    def __init__(self, *args, **kwargs):

        super(Study, self).__init__(*args, **kwargs)
        # self.save() # leads to null value in column "study_type_id" violates not-null constraint, even with study_type defined in the call to G(Study, ...)
        self.machine = Machine(
            self,
            states=workflow.states,
            transitions=workflow.transitions,
            initial=self.state,
            send_event=True,
            before_state_change="check_permission",
            after_state_change="_finalize_state_change",
        )
        # self.save() # With the call to save here, we're back at 'Model instances without primary key value are unhashable'
```
